applications/tari_base_node/src/builder.rs

```diff
@@ -905,14 +905,23 @@ async fn setup_wallet_comms(
 }
 
 async fn add_peers_to_comms(comms: &CommsNode, peers: Vec<Peer>) -> Result<(), String> {
+    let node_identity = comms.node_identity().clone();
     for p in peers {
         let peer_desc = p.to_string();
         info!(target: LOG_TARGET, "Adding seed peer [{}]", peer_desc);
-        comms
-            .peer_manager()
-            .add_peer(p)
-            .await
-            .map_err(|e| format!("Could not add peer {} to comms layer: {}", peer_desc, e))?;
+
+        if p.public_key != node_identity.public_key().to_owned() {
```

Comment:
As clippy says :) - 
```suggestion
        if &p.public_key != node_identity.public_key() {
```

Reply:
Rather check for this case and use `continue` 
```suggestion
        if node_identity.public_key() == &p.public_key {
        info!(
                target: LOG_TARGET,
                "Attempting to add yourself [{}] as a seed peer to comms layer, ignoring request", peer_desc
            );
            continue;
            }
```